### ISSUE TITLE
crash/coredump after errormessage "Too many levels of symbolic links"

### ISSUE THREAD

**Issue by klartext** [neomutt/neomutt#1898]
## Expected Behaviour
Saving mail into folder, even though the folder is a symbolic link to a real file.
And in any case: no crash.

Saving an email into a folder (which is a symbolic link to the real folder-file).
After failure using "q" to quit neomutt.

## Actual Behaviour
Errormessage
```
<path-to-symlink-folder>: Too many levels of symbolic links (errno = 40)
```

When quitting right after that message, I got an errormessage with coredump.

```
<shallscriptname-that-invokes neomutt>: line 3:  3263 Aborted                 (core dumped) neomutt -f <imap-address>
```

crash after quit.

If NeoMutt crashed, did your OS create a 'coredump' file?
Yes.

## Steps to Reproduce
save an email into a folder, which is a symlink to the real folder file.
After you got the errormessage "Too many levels of symbolic links" in the status bar, quit.

## How often does this happen?

- Always

## When did it start to happen?
With upgrade to release NeoMutt 2019-10-25

## NeoMutt Version

```
NeoMutt 2019-10-25
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.3.7-arch1-1-ARCH (x86_64)
ncurses: ncurses 6.1.20180127 (compiled with 6.1.20180127)
libidn2: 2.2.0 (compiled with 2.2.0)
GPGme: 1.13.1
hcache backends: kyotocabinet, gdbm, lmdb

Compiler:
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto
Thread model: posix
gcc version 9.2.0 (GCC) 

Configure options: --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --gpgme --enable-lua --notmuch --gss --ssl --sasl --with-ui=ncurses --with-idn2=/usr --idn=0 --idn2=1 --lmdb --kyotocabinet --gdbm

Compilation CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool +idn +inotify 
  -locales_hack +lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version
see above (neomutt -v)

* Were you using multiple copies of NeoMutt at once?
no.

* Were you using 'screen' or 'tmux'?
no

* Is your email local (maildir) or remote (IMAP)?
email in imap folder, saving local

Btw: too many levels of symbolic links: it's a symlink direclty to the folder, no other symlinks in between.

**Comment by gahr** [comment]
I have been able to reproduce the following behaviours, given

```
drwxr-xr-x  2 gahr  gahr       512 29 Oct 15:28 real-path
lrwxr-xr-x  1 gahr  gahr         9 29 Oct 12:20 sym-path -> real-path
```

| Save to | Result |
| ------- | ------- |
| `./real-path/` | `./real-path: Is a directory (errno = 21)` |
| `./real-path` | crash |
| `./sym-path/` | `./sym-path: Too many links (errno = 31)` |
| `./sym-path` | crash |

**Comment by flatcap** [comment]
Fixed in: ad00c1bf0
